tests/datavolume_test.go

```diff
@@ -119,6 +120,84 @@ var _ = Describe("[vendor:cnv-qe@redhat.com][level:component]DataVolume tests",
 		)
 	})
 
+	Describe("Verify DataVolume with block mode", func() {
+		var err error
+		var storageClass *storageV1.StorageClass
+		var pod *v1.Pod
+		var pv *v1.PersistentVolume
+
+		BeforeEach(func() {
+			By(fmt.Sprintf("Creating storageClass for Block PV"))
+			storageClass, err = f.CreateStorageClassFromDefinition(utils.NewStorageClassForBlockPVDefinition("manual"))
+			Expect(err).ToNot(HaveOccurred())
+
+			pod, err = utils.FindPodByPrefix(f.K8sClient, "cdi", "cdi-block-device", "kubevirt.io=cdi-block-device")
+			Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Unable to get pod %q", "cdi"+"/"+"cdi-block-device"))
+
+			nodeName := pod.Spec.NodeName
+			By(fmt.Sprintf("Creating Block PV"))
+			pv, err = f.CreatePVFromDefinition(utils.NewBlockPVDefinition("local-volume1", "1Gi", nil, "manual", nodeName))
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Verify that PV's phase is Available.")
+			err = f.WaitTimeoutForPVReady(pv.Name, 60*time.Second)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		AfterEach(func() {
+			err := utils.DeletePV(f.K8sClient, pv)
+			Expect(err).ToNot(HaveOccurred())
+
+			err = utils.DeleteStorageClass(f.K8sClient, storageClass)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		table.DescribeTable("should", func(name, command, url, dataVolumeName, eventReason string, phase cdiv1.DataVolumePhase) {
+			var dataVolume *cdiv1.DataVolume
+
+			switch name {
+			case "import-http":
```

Comment:
Don't need this switch for a single case. You can just create the DV because that is the only option you have.

Reply:
I'm leaving it right now as is, since  do not want to revert it back to switch block when I add clone and upload (VERY soon)

Reply:
Sure